obspy/io/csv/core.py

```diff
@@ -0,0 +1,484 @@
+                    raise
+            except Exception:
+                magnitudes = []
+            else:
+                try:
+                    magtype = row['magtype']
+                    if magtype.lower() in ('', 'none', 'null', 'nan'):
+                        raise
+                except Exception:
+                    magtype = default.get('magtype')
+                magauthor = _string(row, 'magauthor')
+                info = (evmod.CreationInfo(author=magauthor) if magauthor
+                        else None)
+                magnitudes = [evmod.Magnitude(
+                    mag=mag, magnitude_type=magtype, creation_info=info)]
+            try:
+                id_ = evmod.ResourceIdentifier(row['id'].strip())
+            except Exception:
+                id_ = None
+            region = _string(row, 'region')
+            descs = ([evmod.EventDescription(region, 'region name')]
+                     if region else [])
+            event = evmod.Event(
+                magnitudes=magnitudes,
+                origins=[origin],
+                resource_id=id_,
+                event_descriptions=descs
+            )
+            events.append(event)
+            if format_check:
+                return True
+    if format_check:
+        # empty file will return an empty catalog,
+        # but it is not detected as CSV file
+        return False
+    return evmod.Catalog(events=events)
+
+
+def _write_csv(events, fname, fields='basic', depth_in_km=True, delimiter=',',
+               header=None):
+    """
+    Write ObsPy catalog to CSV file
+
+    :param events: catalog or list of events
+    :param fname: file name
+    :param depth_in_km: write depth in units of kilometer (default: True) or
+        meter
+    :param delimiter: defaults to `','`, if the delimiter is changed, ObsPy's
+        read_events function will not automatically identify the file as
+        CSV file
+    """
+    fields = FIELDS.get(fields, fields)
+    if ' ' in fields:
+        fields = fields.split()
+    fmtstr = delimiter.join(fields)
+    if not depth_in_km and 'depm' not in fmtstr:
+        fmtstr = fmtstr.replace('dep', 'depm')
+    if header is None:
+        fieldnames = [
+            fn for _, fn, _, _ in Formatter().parse(fmtstr) if fn is not None]
+        header = delimiter.join(fieldnames)
+    with _open(fname, 'w') as f:
+        f.write(header + '\n')
+        for event in events:
+            evid = str(event.resource_id).split('/')[-1]
+            try:
+                origin = _origin(event)
+            except Exception:
+                warn(f'No origin found -> do not write event {evid}')
+                continue
+            try:
+                author = origin.creation_info.author
+            except AttributeError:
+                author = ''
+            try:
+                contrib = origin.creation_info.agency_id
+            except AttributeError:
+                contrib = ''
```

Comment:
These cant be re-phrased as `getattr` with default values because both creation_info and creation_info's sub attribute might be missing right?

Reply:
Exactly